common/service/dynamicconfig/constants.go

```diff
@@ -343,14 +339,15 @@ const (
 	TransactionSizeLimit
 	// MinRetentionDays is the minimal allowed retention days for namespace
 	MinRetentionDays
-	// MaxWorkflowTaskTimeout  is the maximum allowed workflow task start to close timeout
-	MaxWorkflowTaskTimeout
 	// DisallowQuery is the key to disallow query for a namespace
 	DisallowQuery
 	// EnablePriorityTaskProcessor is the key for enabling priority task processor
 	EnablePriorityTaskProcessor
 	// EnableAuthorization is the key to enable authorization for a namespace
 	EnableAuthorization
+	// EnableInfiniteTimeout is the key to enable infinite timeout
+	// TODO remove after 1.5
+	EnableInfiniteTimeout
```

Comment:
Do we really need a feature flag for this?  This does not break any backwards compatibility? right?

Reply:
during deployment, new frontend can make RPC call to old history service, making the workflow timeout 0, meaning timeout the workflow immediately

Reply:
there are other case, like 
1. new history service write down an event, which will be processed by old history service (shard movement)
2. new history service make RPC call to old history service (start child workflow)

Reply:
But the default value for flag is `true`, won't this still run into these issues?

Reply:
good catch, default should be false